### PR TITLE
compress all deps which are noted in the schema to archive

### PR DESCRIPTION
**Pull request by 0xAX**

### CODE REVIEW

File: lib/exrm/plugins/conform.ex [hunk 1]
```diff
@@ -34,6 +34,24 @@ defmodule ReleaseManager.Plugin.Conform do
...
+          {:ok, files_list} = :file.list_dir("#{deps_app}/ebin")
+          files_list = Enum.map(files_list, fn(filename) -> ("#{deps_app}/ebin/" <> "#{filename}") |> to_char_list end)
+          :lists.append(acc, files_list)
+        end)
+
+        case build_files do
+          [] ->
+            File.cd(curr_path)
+          _ ->
+            :zip.create("#{curr_path}/rel/#{app}/releases/#{version}/app.schema.deps.ez" |> to_char_list, build_files)
```
Comment (liveforeverx) [inline lib/exrm/plugins/conform.ex]: I think, it would be better to create zip archive in rel/files (as schema) and add it to overlays, as schema.

Comment (bitwalker) [inline lib/exrm/plugins/conform.ex]: Yeah I agree with @liveforeverx, should be able to do this by doing something like this:

```
# Add relx_config to pattern match in `before_release` head
def before_release(%Config{name: app, version: version, relx_config: relx_config}) do

# Then create the zip and add the overlay
zip_path         = :zip.create(...)
with_overlays = [{:overlay, [{:copy, '#{zip_path}', 'releases/#{version}/app.schema.deps.ez'}]}]
relx_config     = Utils.merge(relx_config, with_overlays)

# Persist the updated relx.config
Utils.write_terms(Utils.rel_file_dest_path("relx.config"), relx_config)

# Make sure you return the updated config to exrm
%{config | :relx_config => relx_config}
```

**Comment by 0xAX** [comment]
@bitwalker Tests are broken because they depends on conform functional from the - https://github.com/bitwalker/conform/pull/33

I've tested it with new changes in conform and all tests are passed sucessfully. Let me know if need to change something.

**Comment by bitwalker** [comment]
I can merge this once the PR in conform is ready, just have a few questions in my comment there :)

File: test/fixtures/fake_project/mix.exs [hunk 2]
```diff
@@ -0,0 +1,30 @@
+defmodule FakeProject.Mixfile do
+  use Mix.Project
+
+  def project do
+    [app: :fake_project,
+     version: "0.0.1",
+     elixir: "~> 1.1-dev",
```
Comment (bitwalker) [inline test/fixtures/fake_project/mix.exs]: This fails to compile in Travis. Could you change the Elixir requirement to `~> 1.0`?

**Comment by 0xAX** [comment]
hmm weird, last build broken, will look

**Comment by 0xAX** [comment]
Hello @bitwalker, I've analyzed & tested today the `exrm` with the different versions of the elixir compiler today . The first is the elixir 1.0 (as you asked). If I'll run tests with this version of elixirlang, I'm getting the following problem:

```elixir
...
...
...
==> conform
could not compile dependency conform, mix compile failed. You can recompile this dependency with `mix deps.compile conform` or update it with `mix deps.update conform`
** (UndefinedFunctionError) undefined function: :neotoma.file/1 (module :neotoma is not available)
    :neotoma.file('/home/alex/work/exrm/deps/conform/src/conf_parse.peg')
    tasks/compile.peg.exs:26: Mix.Tasks.Compile.Peg.compile_peg/1
    (elixir) lib/enum.ex:977: Enum."-map/2-lc$^0/1-0-"/2
    (mix) lib/mix/tasks/compile.ex:59: anonymous fn/2 in Mix.Tasks.Compile.run/1
    (elixir) lib/enum.ex:977: Enum."-map/2-lc$^0/1-0-"/2
    (mix) lib/mix/tasks/compile.ex:58: Mix.Tasks.Compile.run/1
    (mix) lib/mix/tasks/deps.compile.ex:104: anonymous fn/2 in Mix.Tasks.Deps.Compile.do_mix/1
    (mix) lib/mix/project.ex:194: Mix.Project.in_project/4

  1) test can build a release and boot it up (UtilsTest)
     /home/alex/work/exrm/test/utils_test.exs:67
     match (=) failed
     code: :ok = Utils.mix("do deps.get, compile", Mix.env(), :verbose)
     rhs:  {:error, "Release step failed. Please fix any errors and try again."}
     stacktrace:
       test/utils_test.exs:70josev

  2) test can fetch a list of plugins (PluginTest)
     /home/alex/work/exrm/test/plugin_test.exs:5
     match (=) failed
     code: :ok = Utils.mix("do deps.get, compile", Mix.env(), :quiet)
     rhs:  {:error, "Release step failed. Please fix any errors and try again."}
     stacktrace:
       test/plugin_test.exs:11
```

If I'll clean fixtures, it works first time but after this I'm getting errors related to the neotoma. 
Also I've tried to test it with the elixir from master, but there is:

```elixir
....
....
....
e {rlx_app_discovery,
      [{unversioned_app,
           <<"/home/alex/work/elixir/bin/../lib/mix/test/fixtures/archive">>,
           local_sample},
       {unversioned_app,
           <<"/home/alex/work/elixir/lib/mix/test/fixtures/archive">>,
           local_sample}]}
==> Failed to build release. Please fix any errors and try again.


  1) test can build a release and boot it up (UtilsTest)
     /home/alex/work/exrm/test/utils_test.exs:67
     match (=) failed
     code: :ok = Utils.mix("release", Mix.env())
     rhs:  {:error, "Release step failed. Please fix any errors and try again."}
     stacktrace:
       test/utils_test.exs:71

.........

Finished in 16.1 seconds (0.1s on load, 16.0s on tests)
12 tests, 1 failure
```

As you can see the problem is here `elixir/lib/mix/test/fixtures/archive`. If we add versions to the `lib/mix/test/fixtures/archive` project in the elixir source code, all tests will be successful:

```
Finished in 22.3 seconds (0.1s on load, 22.1s on tests)
12 tests, 0 failures
```

Some days ago I've sent pull request to the elixir (https://github.com/elixir-lang/elixir/pull/3526) but @josevalim said that the problem is in other place.

What do you think about this?

**Comment by liveforeverx** [comment]
@0xAX, hey, we have the same problem here: #178 

Second problem is fixed on exrm master, please, rebase.

**Pull request closed without merging**